### PR TITLE
Implementing the faithful representation of a generic Lie algebra in positive characteristic

### PR DESCRIPTION
**Pull request by tscrim**
This is a very large representation in general, but it is known to be faithful.

### :memo: Checklist

- [x] The title is concise and informative.
- [x] The description explains in detail what this PR is about.
- [x] I have linked a relevant issue or discussion.
- [x] I have created tests covering the changes.
- [x] I have updated the documentation accordingly.

### :hourglass: Dependencies

- #37724 Builds on this

### CODE REVIEW

**Comment by mkoeppe** [comment]
lint failure

**Comment by tscrim** [comment]
I've addressed the lint failure.

File: src/sage/algebras/lie_algebras/representation.py [hunk 1]
```diff
@@ -791,3 +791,242 @@ def _acted_upon_(self, scalar, self_on_left=False):
...
+    minimal polynomial of `A`. The `(f_1, \ldots, f_n)` is a Gröbner basis
+    for an ideal `I` of the universal enveloping algebra `U(L)` such that
+    the quotient `U(L) / I` is a faithful representation of `L`.
+
+    EXAMPLES::
+
+        sage: sl2 = LieAlgebra(GF(3), cartan_type=['A',1])
+        sage: F = sl2.faithful_representation()
+        sage: F
+        Faithful representation with p-multiplicies (1, 3, 1) of Lie algebra
```
Comment (mkoeppe) [inline src/sage/algebras/lie_algebras/representation.py]: s/multiplicies/multiplicities/g

File: src/sage/algebras/lie_algebras/representation.py [hunk 2]
```diff
@@ -791,3 +791,242 @@ def _acted_upon_(self, scalar, self_on_left=False):
...
+
+        This is often a very large dimensional representation relative
+        to the dimension of the Lie algebra.
+
+    ALGORITHM:
+
+    We implement the algorithm given in [deG2000] Section 6.6. Let `L`
+    be a finite dimensional Lie algebra over a ring of characteristic `p`
+    with basis `(b_1, \ldots, b_n)`. We compute (monic) `p`-polynomials
+    `f_i` such that `A = \mathrm{ad}(b_i)` solves `f_i(A) = 0` by using
```
Comment (mkoeppe) [inline src/sage/algebras/lie_algebras/representation.py]: Need to explain what "ad" is?

Comment (tscrim) [reply src/sage/algebras/lie_algebras/representation.py]: I explained a bit but I just used the full terminology. I expect anyone reading this should know what the adjoint action is, or at least can very quickly find it.

Comment (mkoeppe) [reply src/sage/algebras/lie_algebras/representation.py]: Thanks, that's good enough

**Comment by tscrim** [comment]
This was working for me. In the last beta version, the import was removed. Updated and fixed.

**Review by mkoeppe — APPROVED** [review]
LGTM. Test failure is unrelated.

**Comment by tscrim** [comment]
Thank you!

**Pull request merged**